be/src/exprs/aggregate_functions.cpp

```diff
@@ -231,6 +232,84 @@ void AggregateFunctions::avg_update(FunctionContext* ctx, const T& src, StringVa
     ++avg->count;
 }
 
+struct PercentileApproxState {
+public:
+    TDigest *digest = nullptr;
+    double targetQuantile = -1.0;
+
+    ~PercentileApproxState() {
+        if (digest != nullptr) {
+            delete digest;
+        }
+    }
+};
+
+void AggregateFunctions::percentile_approx_init(FunctionContext* ctx, StringVal* dst) {
+    dst->is_null = false;
+    dst->len = sizeof(PercentileApproxState);
+    dst->ptr = (uint8_t*) new PercentileApproxState();
+
+    PercentileApproxState* percentile = reinterpret_cast<PercentileApproxState*>(dst->ptr);
+    percentile->targetQuantile = -1.0;
+    percentile->digest = new TDigest();
```

Comment:
I think you'd better to move `new TDigest()` to PercentileApproxState's constructor